client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx

```diff
@@ -0,0 +1,239 @@
+		const newSuggestions = verticals.filter( ( vertical ) =>
+			vertical.label.toLowerCase().includes( normalizedInputValue )
+		);
+
+		// Does the verticals list include an exact match?
+		// If yes, we store it in firstSuggestion (for later use), and remove it from newSuggestions...
+		const firstSuggestion = remove(
+			newSuggestions,
+			( suggestion ) => suggestion.label.toLowerCase() === normalizedInputValue
+		)[ 0 ] ?? {
+			// ...otherwise, we set firstSuggestion to the user-supplied vertical...
+			label: inputValue.trim(),
+			id: '', // User-supplied verticals don't have IDs or slugs
+			slug: '',
+		};
+
+		// ...and finally, we prepend firstSuggestion to our suggestions list.
+		newSuggestions.unshift( firstSuggestion );
+
+		setSuggestions( newSuggestions );
+	};
+
+	const handleSelect = ( vertical: SiteVertical ) => {
+		setSiteVertical( vertical );
+		setIsFocused( false ); // prevent executing handleBlur()
+		// empty suggestions cache once a vertical is selceted
+		setSuggestions( [] );
+	};
+
+	const selectLastInputValue = () => {
+		const lastQuery = inputText.trim();
+		if ( isFocused && lastQuery.length ) {
+			const vertical = suggestions[ 0 ] ?? { label: lastQuery, id: '', slug: '' };
+			handleSelect( vertical );
+		}
+	};
+
+	const handleSuggestAction = ( vertical: SiteVertical ) => {
+		handleSelect( vertical );
+		onNext();
+	};
+
+	const handleInputKeyUpEvent = (
+		e: React.KeyboardEvent< HTMLInputElement > | React.ChangeEvent< HTMLInputElement >
+	) => {
+		const input = e.currentTarget.value.trim();
+		if ( ! input.length ) {
+			resetSiteVertical();
+		}
+		updateSuggestions( input );
+	};
+
+	const handleInputKeyDownEvent = ( e: React.KeyboardEvent< HTMLInputElement > ) => {
+		const input = e.currentTarget.value.trim();
+
+		if ( suggestionRef.current ) {
+			suggestionRef.current.handleKeyEvent( e );
+		}
+		if ( e.keyCode === ENTER ) {
+			e.preventDefault();
+			input.length && ! suggestions.length && handleSelect( { label: input } );
+			return;
+		}
+		if ( e.keyCode === TAB && input.length ) {
+			e.preventDefault();
+			selectLastInputValue();
+		}
+	};
+
+	React.useEffect( () => {
+		if ( isInputEmpty ) {
+			inputRef?.current?.focus();
+		}
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	React.useEffect( () => {
+		const { slug, label } = siteVertical || {};
+		setTextValue( label ?? '' );
+		recordVerticalSelection( slug, label );
+	}, [ siteVertical ] );
+
+	React.useEffect( () => {
+		if ( isMobile ) {
+			window.scrollTo( 0, 0 );
+		}
+	}, [ suggestions, isMobile ] );
+
+	return (
+		/* eslint-disable jsx-a11y/click-events-have-key-events */
+		/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+		<form
+			className={ classnames( 'vertical-select', {
+				'vertical-select--with-suggestions': !! suggestions.length && isMobile,
+			} ) }
+		>
+			<label htmlFor="vertical-input">{ __( 'My site is about' ) } </label>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'My site is about <Input />.' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/vertical-input)